### ISSUE TITLE
Add base path to items in Sidebar

### ISSUE THREAD

**Issue by tvld** [vuejs/vitepress#1158]
### Is your feature request related to a problem? Please describe.

We need to duplicate the folder tree in the ``link: '...'`` for each document in the sidebar. Even the [documentation](https://vitepress.vuejs.org/guide/theme-sidebar.html) shows that.

### Describe the solution you'd like

I would suggest adding an optional parameter. So, Instead of:

 ```js
{
    text: 'Inclusivity',
    items: [
      { text: 'Employee Handbook', link: '/inclusive/employee-handbook' },
      { text: 'Not a Wasp', link: '/inclusive/not-a-wasp' },
    ]
  }
```

use this 
 ```js {2}
{
    text: 'Inclusivity',
    basePath: '/inclusive/',
    items: [
      { text: 'Employee Handbook', link: 'employee-handbook' },
      { text: 'Not a Wasp', link: 'not-a-wasp' },
    ]
  }
```

### Validations

- [X] Follow our [Code of Conduct](https://vuejs.org/about/coc.html)
- [X] Read the [docs](https://vitepress.vuejs.org).
- [X] Read the [Contributing Guidelines](https://github.com/vuejs/vitepress/blob/main/.github/contributing.md).
- [X] Check that there isn't already an issue that asks for the same feature to avoid creating a duplicate.

**Comment by xsjcTony** [comment]
@brc-dd The current implementation will break both link and highlights, as well as the footer prevNext navigation if `base` is used on item with `link` together.

e.g.
```ts
    sidebar: {
      '/': [
        {
          text: 'Root with link',
          link: '/abc/', // (Using `/` here will make the link work, but highlighting and prevNext will still break)
          base: '/abc/
          items: [
            {
              text: 'Child',
              link: 'child',
            },
          ],
        },
      ],
    },

**Comment by brc-dd** [comment]
what about `link: '/', base: '/abc'`?

**Comment by brc-dd** [comment]
They shouldn't break with the base I suggested (notice no trailing slash after `abc`). base is simply added to link in pre processing. It doesn't has any special meaning.

**Comment by xsjcTony** [comment]
Yeah that's making sense. I didn't notice there's not trailing `/` in the base. Although it's a pity that the base can't be concatenated which makes multi-level nested links a little bit hard to work with, but it's still an impressive enhancement💚
